src/beach.ts

```diff
@@ -0,0 +1,47 @@
+import { cliExecute, fileToBuffer, gamedayToInt, myId } from "kolmafia";
+import { get, set } from "libram";
+
+type BeachTile = { minute: number; row: number; column: number };
+
+const rareTiles: BeachTile[] = JSON.parse(fileToBuffer("raretiles.json"));
+
+function _comb(tile: BeachTile): void {
+  const { minute, row, column } = tile;
+  cliExecute(`beach wander ${minute}; beach comb ${row} ${column}`);
+}
+
+let seed = parseInt(myId());
+function deterministicRandom(): number {
+  seed++;
+  return Math.sin(seed);
+}
+
+function deterministicShuffle<T>(array: T[]): T[] {
+  const returnValue: T[] = [];
+  while (array.length > 0) {
+    const index = Math.floor(deterministicRandom() * array.length);
+    returnValue.push(...array.splice(index));
+  }
+  return returnValue;
+}
+
+let shuffledArray: BeachTile[];
+
+function getShuffledArray(): BeachTile[] {
+  if (!shuffledArray) {
+    shuffledArray = deterministicShuffle(rareTiles);
+  }
+  return shuffledArray;
+}
+
+export default function comb(): void {
+  const tileList = getShuffledArray();
+  const index = (get("garbo_lastTileCombed", 0) + 1) % tileList.length;
+  const tile = tileList[index];
+  const dayOfMonth = 1 + (gamedayToInt() % 8);
```

Comment:
I'm also going to assume that gamedayToInt works the way you intend it to. I assume you've tested that the beach data this gives matches what we are observing from logs etc.

Reply:
This is code stolen from SSBBHax, who thought that linking tide patterns to the game month was a hacky solution instead of a great one. gamedayToInt does; today is day 83, and it's Boozember 4. We want to get 4 out of it, so this code should do that.